### PR TITLE
feat: Remove react resize detector

### PR DESCRIPTION
**Pull request by lee-chase**
Contributes to #2744 

V7 of ReactResizeObserver can result in the error shown in #2744 but the error appears to be a timing issue of some description.

The author fixed a similar looking issue in V8, but it persists in our storybook. In V9 there were more extensive changes that stopped our current usage working.

This PR removes ReactResizeObserver in favor of wrapping ResizeObserver (now standard) in a hook.

#### What did you change?

- Removed react-resize-detector
- Created useResizeObserver hook.
- Checked it functioned as a replacement for all useResizeDetector hook

#### How did you test and verify your work?

- Storybook changing the size of things and adding new elements
- Ran tests

NOTE: Storybook 6 reports the following error which would appear to be related this issue fixed only in Storybook 7. https://github.com/storybookjs/storybook/issues/18716

image

### CODE REVIEW

File: packages/cloud-cognitive/src/global/js/hooks/useWindowScroll.js [hunk 1]
```diff
@@ -1,3 +1,9 @@
+/**
+ * Copyright IBM Corp. 2022, 2022
```
Comment (elycheea) [inline packages/cloud-cognitive/src/global/js/hooks/useWindowScroll.js:2]:
```suggestion
 * Copyright IBM Corp. 2022, 2023
```

File: packages/cloud-cognitive/src/global/js/hooks/useWindowResize.js [hunk 2]
```diff
@@ -1,3 +1,9 @@
+/**
+ * Copyright IBM Corp. 2022, 2022
```
Comment (elycheea) [inline packages/cloud-cognitive/src/global/js/hooks/useWindowResize.js:2]:
```suggestion
 * Copyright IBM Corp. 2022, 2023
```

File: packages/cloud-cognitive/src/components/SidePanel/SidePanel.js [hunk 3]
```diff
@@ -169,7 +169,8 @@ export let SidePanel = React.forwardRef(
     }, [labelText, title]);
 
     /* istanbul ignore next */
-    const handleResize = (width, height) => {
+    const handleResize = ({ height }) => {
+      console.log(height);
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/SidePanel/SidePanel.js:173]: Remove console log

File: packages/cloud-cognitive/src/components/PageHeader/PageHeader.js [hunk 4]
```diff
@@ -162,10 +162,10 @@ export let PageHeader = React.forwardRef(
     };
 
     /* istanbul ignore next */
-    const handleResizeActionBarColumn = (width) => {
+    const handleResizeActionBarColumn = ({ width }) => {
       /* don't know how to test resize */
       /* istanbul ignore next */
-      setActionBarColumnWidth(width);
+      setActionBarColumnWidth({ width });
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/PageHeader/PageHeader.js:168]: I don't know as much about the PageHeader, but how come you've changed how the `actionBarColumnWidth` is set?

ie:
`setActionBarColumnWidth(width);`
vs
`setActionBarColumnWidth({ width });`

It looks like `width` is already destructured.

Comment (lee-chase) [reply packages/cloud-cognitive/src/components/PageHeader/PageHeader.js]: Fixed, good catch.

File: packages/cloud-cognitive/src/components/Tearsheet/TearsheetShell.js [hunk 5]
```diff
@@ -102,8 +102,11 @@ export const TearsheetShell = React.forwardRef(
     }, [portalTargetIn]);
 
     const localRef = useRef();
+    const resizer = useRef(null);
     const modalRef = ref || localRef;
-    const { width, ref: resizer } = useResizeDetector({ handleHeight: false });
+    const { width } = useResizeObserver(resizer, (size) =>
```
[the remark this answers is not in the document]
Comment (lee-chase) [reply packages/cloud-cognitive/src/components/Tearsheet/TearsheetShell.js]: Removed console log, thnaks.

File: packages/cloud-cognitive/src/components/ActionBar/ActionBar.js [hunk 6]
```diff
@@ -187,33 +187,20 @@ export let ActionBar = React.forwardRef(
...
-      targetRef: sizingRef,
-    });
-
-    const { ref: outerRef } = useResizeDetector({
-      onResize: handleResize,
-      targetRef: ref,
-    });
+    // // resize of the items
+    useResizeObserver(sizingRef, { callback: handleResize });
+    useResizeObserver(ref, { callback: handleResize });
```
[the remark this answers is not in the document]
Comment (lee-chase) [reply packages/cloud-cognitive/src/components/ActionBar/ActionBar.js]: Good point, fixed.

**Review by matthewgallo — APPROVED** [review]
👏 looks great! And nice to have one less dependency.

**Comment by tonydiaz** [comment]
@lee-chase Can this also be removed in the `main` branch? I seeing this issue on `@carbon/ibm-cloud-cognitive@1.51.0`

**Pull request merged**